integration/tests/cook/test_basic.py

```diff
@@ -2465,8 +2465,44 @@ def test_pools_in_default_limit_response(self):
                 resp = util.get_limit(self.cook_url, limit, user)
                 self.assertFalse('pools' in resp.json())
 
+    @pytest.mark.serial
     def test_data_local_support(self):
-        uuid, resp = util.submit_job(self.cook_url, data_local=True)
+        max_cost = util.settings(self.cook_url)['data-local-fitness-calculator']['maximum-cost']
+        slaves = util.get_mesos_state(self.mesos_url)['slaves']
+        costs = []
+        for slave in slaves:
+            costs.append({'node': slave['hostname'], 'cost': max_cost})
+        costs[0]['cost'] = 0
```

Comment:
Can we grade the costs: `0.1 * max-cost`, `0.2 * max-cost` and so on? The remaining parts and assertions of the test don't need to change.